File: app/src/main/java/com/pedrosoares/cotacaoapp/model/mapper/CoinsMapper.java
package com.pedrosoares.cotacaoapp.model.mapper;

import android.support.annotation.NonNull;

import com.pedrosoares.cotacaoapp.data.entity.ARS;
import com.pedrosoares.cotacaoapp.data.entity.BTC;
import com.pedrosoares.cotacaoapp.data.entity.CoinsResponse;
import com.pedrosoares.cotacaoapp.data.entity.ETH;
import com.pedrosoares.cotacaoapp.data.entity.EUR;
import com.pedrosoares.cotacaoapp.data.entity.GBP;
import com.pedrosoares.cotacaoapp.data.entity.LTC;
import com.pedrosoares.cotacaoapp.data.entity.USD;
import com.pedrosoares.cotacaoapp.data.entity.XRP;
import com.pedrosoares.cotacaoapp.model.domain.ARSDomain;
import com.pedrosoares.cotacaoapp.model.domain.BTCDomain;
import com.pedrosoares.cotacaoapp.model.domain.CoinsDomain;
import com.pedrosoares.cotacaoapp.model.domain.ETHDomain;
import com.pedrosoares.cotacaoapp.model.domain.EURDomain;
import com.pedrosoares.cotacaoapp.model.domain.GBPDomain;
import com.pedrosoares.cotacaoapp.model.domain.LTCDomain;
import com.pedrosoares.cotacaoapp.model.domain.USDDomain;
import com.pedrosoares.cotacaoapp.model.domain.XRPDomain;

public class CoinsMapper {

    public static CoinsDomain transformEntityToDomain(CoinsResponse coinsResponse) {

        CoinsDomain coinsResponseDomain = new CoinsDomain();
        if (coinsResponse != null){
           coinsResponseDomain.setUSD(transformObjectEntityToDomain(coinsResponse.getUSD()));
           coinsResponseDomain.setEUR(transformObjectEntityToDomain(coinsResponse.getEUR()));
           coinsResponseDomain.setBTC(transformObjectEntityToDomain(coinsResponse.getBTC()));
           coinsResponseDomain.setLTC(transformObjectEntityToDomain(coinsResponse.getLTC()));
           coinsResponseDomain.setGBP(transformObjectEntityToDomain(coinsResponse.getGBP()));
           coinsResponseDomain.setARS(transformObjectEntityToDomain(coinsResponse.getARS()));
           coinsResponseDomain.setXrp(transformObjectEntityToDomain(coinsResponse.getXrp()));
           coinsResponseDomain.setEth(transformObjectEntityToDomain(coinsResponse.getEth()));
        }

        return coinsResponseDomain;
    }

    private static USDDomain transformObjectEntityToDomain(@NonNull USD usdEntity) {
        USDDomain usdDomain = new USDDomain();

        usdDomain.setName(usdEntity.getName());
        usdDomain.setAsk(usdEntity.getAsk());
        usdDomain.setLow(usdEntity.getLow());
        usdDomain.setHigh(usdEntity.getHigh());

        return usdDomain;
    }

    private static EURDomain transformObjectEntityToDomain(@NonNull EUR eurEntity) {
        EURDomain eurDomain = new EURDomain();

        eurDomain.setName(eurEntity.getName());
        eurDomain.setAsk(eurEntity.getAsk());
        eurDomain.setLow(eurEntity.getLow());
        eurDomain.setHigh(eurEntity.getHigh());

        return eurDomain;
    }

    private static BTCDomain transformObjectEntityToDomain(@NonNull BTC btcEntity) {
        BTCDomain btcDomain = new BTCDomain();

        btcDomain.setName(btcEntity.getName());
        btcDomain.setAsk(btcEntity.getAsk());
        btcDomain.setLow(btcEntity.getLow());
        btcDomain.setHigh(btcEntity.getHigh());
        btcDomain.setCreateDate(btcEntity.getCreateDate());

        return btcDomain;
    }

    private static LTCDomain transformObjectEntityToDomain(@NonNull LTC ltcEntity) {
        LTCDomain ltcDomain = new LTCDomain();

        ltcDomain.setName(ltcEntity.getName());
        ltcDomain.setAsk(ltcEntity.getAsk());
        ltcDomain.setLow(ltcEntity.getLow());
        ltcDomain.setHigh(ltcEntity.getHigh());

        return ltcDomain;
    }

    private static GBPDomain transformObjectEntityToDomain(@NonNull GBP gbpEntity) {
        GBPDomain gbpDomain = new GBPDomain();

        gbpDomain.setName(gbpEntity.getName());
        gbpDomain.setAsk(gbpEntity.getAsk());
        gbpDomain.setLow(gbpEntity.getLow());
        gbpDomain.setHigh(gbpEntity.getHigh());

        return gbpDomain;
    }

    private static ARSDomain transformObjectEntityToDomain(@NonNull ARS arsEntity) {
        ARSDomain arsDomain = new ARSDomain();

        arsDomain.setName(arsEntity.getName());
        arsDomain.setAsk(arsEntity.getAsk());
        arsDomain.setLow(arsEntity.getLow());
        arsDomain.setHigh(arsEntity.getHigh());

        return arsDomain;
    }

    private static XRPDomain transformObjectEntityToDomain(@NonNull XRP xrpEntity) {
        XRPDomain xrpDomain = new XRPDomain();

        xrpDomain.setName(xrpEntity.getName());
        xrpDomain.setAsk(xrpEntity.getAsk());
        xrpDomain.setLow(xrpEntity.getLow());
        xrpDomain.setHigh(xrpEntity.getHigh());

        return xrpDomain;
    }

    private static ETHDomain transformObjectEntityToDomain(@NonNull ETH ethEntity) {
        ETHDomain ethDomain = new ETHDomain();

        ethDomain.setName(ethEntity.getName());
        ethDomain.setAsk(ethEntity.getAsk());
        ethDomain.setLow(ethEntity.getLow());
        ethDomain.setHigh(ethEntity.getHigh());

        return ethDomain;
    }
}

File: app/src/main/java/com/pedrosoares/cotacaoapp/model/mapper/CoinsMapper.kt
package com.pedrosoares.cotacaoapp.model.mapper

import com.pedrosoares.cotacaoapp.data.entity.*
import com.pedrosoares.cotacaoapp.model.domain.*

object CoinsMapper {
    fun transformEntityToDomain(coinsResponse: CoinsResponse?): CoinsDomain {
        val coinsResponseDomain = CoinsDomain()
        coinsResponse?.let{
            coinsResponseDomain.usd = transformObjectEntityToDomain(it.usd!!)
            coinsResponseDomain.eur = transformObjectEntityToDomain(it.eur!!)
            coinsResponseDomain.btc = transformObjectEntityToDomain(it.btc!!)
            coinsResponseDomain.ltc = transformObjectEntityToDomain(it.ltc!!)
            coinsResponseDomain.gbp = transformObjectEntityToDomain(it.gbp!!)
            coinsResponseDomain.ars = transformObjectEntityToDomain(it.ars!!)
            coinsResponseDomain.xrp = transformObjectEntityToDomain(it.xrp!!)
            coinsResponseDomain.eth = transformObjectEntityToDomain(it.eth!!)
        }
        return coinsResponseDomain
    }

    private fun transformObjectEntityToDomain(usdEntity: USD): USDDomain {
        val usdDomain = USDDomain()
        usdDomain.name = usdEntity.name
        usdDomain.bid = usdEntity.bid
        usdDomain.low = usdEntity.low
        usdDomain.high = usdEntity.high
        return usdDomain
    }

    private fun transformObjectEntityToDomain(eurEntity: EUR): EURDomain {
        val eurDomain = EURDomain()
        eurDomain.name = eurEntity.name
        eurDomain.bid = eurEntity.bid
        eurDomain.low = eurEntity.low
        eurDomain.high = eurEntity.high
        return eurDomain
    }

    private fun transformObjectEntityToDomain(btcEntity: BTC): BTCDomain {
        val btcDomain = BTCDomain()
        btcDomain.name = btcEntity.name
        btcDomain.bid = btcEntity.bid
        btcDomain.low = btcEntity.low
        btcDomain.high = btcEntity.high
        btcDomain.createDate = btcEntity.createDate
        return btcDomain
    }

    private fun transformObjectEntityToDomain(ltcEntity: LTC): LTCDomain {
        val ltcDomain = LTCDomain()
        ltcDomain.name = ltcEntity.name
        ltcDomain.bid = ltcEntity.bid
        ltcDomain.low = ltcEntity.low
        ltcDomain.high = ltcEntity.high
        return ltcDomain
    }

    private fun transformObjectEntityToDomain(gbpEntity: GBP): GBPDomain {
        val gbpDomain = GBPDomain()
        gbpDomain.name = gbpEntity.name
        gbpDomain.bid = gbpEntity.bid
        gbpDomain.low = gbpEntity.low
        gbpDomain.high = gbpEntity.high
        return gbpDomain
    }

    private fun transformObjectEntityToDomain(arsEntity: ARS): ARSDomain {
        val arsDomain = ARSDomain()
        arsDomain.name = arsEntity.name
        arsDomain.bid = arsEntity.bid
        arsDomain.low = arsEntity.low
        arsDomain.high = arsEntity.high
        return arsDomain
    }

    private fun transformObjectEntityToDomain(xrpEntity: XRP): XRPDomain {
        val xrpDomain = XRPDomain()
        xrpDomain.name = xrpEntity.name
        xrpDomain.bid = xrpEntity.bid
        xrpDomain.low = xrpEntity.low
        xrpDomain.high = xrpEntity.high
        return xrpDomain
    }

    private fun transformObjectEntityToDomain(ethEntity: ETH): ETHDomain {
        val ethDomain = ETHDomain()
        ethDomain.name = ethEntity.name
        ethDomain.bid = ethEntity.bid
        ethDomain.low = ethEntity.low
        ethDomain.high = ethEntity.high
        return ethDomain
    }
}